bin/docker-to-squash.py

```diff
@@ -0,0 +1,1430 @@
+      if os.path.isdir(working_dir):
+        shutil.rmtree(working_dir)
+
+def add_remove_tag(args):
+  pull_format = args.pull_format
+  hdfs_root = args.hdfs_root
+  image_tag_to_hash = args.image_tag_to_hash
+  replication = args.replication
+  sub_command = args.sub_command
+  images_and_tags = args.images_and_tags
+
+  hdfs_manifest_dir = hdfs_root + "/manifests"
+  working_dir = None
+
+  try:
+    working_dir = get_working_dir(args.working_dir)
+    local_image_tag_to_hash = os.path.join(working_dir, os.path.basename(image_tag_to_hash))
+    hash_to_tags, tag_to_hash, image_tag_to_hash_hash = populate_tag_dicts(hdfs_root,
+                                                                           image_tag_to_hash,
+                                                                           local_image_tag_to_hash)
+
+    for image_and_tag_arg in images_and_tags:
+      if sub_command == "add-tag":
+        image, tags = split_image_and_tag(image_and_tag_arg)
+        if is_sha256_hash(image):
+          manifest_hash = image
+        else:
+          manifest_hash = tag_to_hash.get(image, None)
+
+        if manifest_hash:
+          manifest_path = hdfs_manifest_dir + "/" + manifest_hash
+          out, err, returncode = hdfs_cat(manifest_path)
+          manifest = json.loads(out)
+          logging.debug("image tag exists for %s", image)
+        else:
+          manifest, manifest_hash = get_manifest_from_docker_image(pull_format, image)
+
+        update_dicts_for_multiple_tags(hash_to_tags, tag_to_hash, tags,
+                                       manifest_hash, image)
+
+      elif sub_command == "remove-tag":
+        tags = image_and_tag_arg.split(",")
+        image = None
+        manifest = None
+        manifest_hash = 0
+        remove_from_dicts(hash_to_tags, tag_to_hash, tags)
+      else:
+        raise Exception("Invalid sub_command: %s" % (sub_command))
+
+    write_local_image_tag_to_hash(local_image_tag_to_hash, hash_to_tags)
+    atomic_upload_mv_to_hdfs(local_image_tag_to_hash, image_tag_to_hash,
+                                     hdfs_root, replication,
+                                     image_tag_to_hash_hash)
+  finally:
+    if working_dir:
+      if os.path.isdir(working_dir):
+        shutil.rmtree(working_dir)
+
+def copy_update(args):
+  image_tag_to_hash = args.image_tag_to_hash
+  replication = args.replication
+  force = args.force
+  src_root = args.src_root
+  dest_root = args.dest_root
+  images_and_tags = args.images_and_tags
+  bootstrap = args.bootstrap
+
+  src_layers_dir = src_root + "/layers"
+  src_config_dir = src_root + "/config"
+  src_manifest_dir = src_root  + "/manifests"
+  dest_layers_dir = dest_root + "/layers"
+  dest_config_dir = dest_root + "/config"
+  dest_manifest_dir = dest_root  + "/manifests"
+
+  if bootstrap:
+    hdfs_dirs = [dest_root, dest_layers_dir, dest_config_dir, dest_manifest_dir]
+    image_tag_to_hash_path = hdfs_root + "/" + image_tag_to_hash
```

Comment:
replace with args.hdfs_root or assign hdfs_root = args.hdfs_root like in some methods before.

Reply:
This should be `dest_root`. Fixed and tested copy_update method.